pyccel/parser/semantic.py

```diff
@@ -1339,6 +1339,9 @@ def _ensure_target(self, rhs, d_lhs):
         if isinstance(rhs, IndexedElement) and rhs.rank > 0 and rhs.base.internal_variable.allocatable:
             d_lhs['allocatable'] = False
             d_lhs['is_pointer' ] = True
+            if rhs.args[0].rank != len(rhs.args) - 1 or any(isinstance(a, Slice) for a in rhs.args):
+                d_lhs['is_view' ] = True
```

Comment:
I don't understand what is being checked in `rhs.args[0].rank != len(rhs.args) - 1` is there an object with a clearer name than `rhs.args[0]`? Is it `rhs.base.internal_variable`?